nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -777,6 +805,14 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path], selective_restore:
         """
         torch.cuda.empty_cache()
 
+        if HAVE_MODELOPT:
+            # If present, first restore and modify the model according to the ModelOpt state.
+            restore_sharded_modelopt_state(
+                [unwrap_model(self.megatron_parallel)],
+                ckpt_to_weights_subdir(checkpoint_path, is_saving=False),
+            )
+            logging.info("Restored Model-Optimizer state from checkpoint.")
```

Comment:
we should log this message only if there was modelopt state in the checkpoint i.e. (`if ModeloptStateManager.is_converted(core_model)`)